attestation/commandrun/tracing_linux.go

```diff
@@ -74,54 +73,104 @@ func (r *CommandRun) trace(c *exec.Cmd, actx *attestation.AttestationContext) ([
 	return pctx.procInfoArray(), nil
 }
 
-func (p *ptraceContext) runTrace() error {
+func (p *ptraceContext) runTrace(ctx context.Context) error {
+	// Create a new channel for reporting errors.
+	errChan := make(chan error)
```

Comment:
this is all synchronous code, we don't really need a channel and go routine to print logging messages here, calling log.Errorf at the erros site is easier and adds less cognitive burden when reasoning about what's going on with these errors.